content/ember/v3/array-observers.md

```diff
@@ -0,0 +1,345 @@
+
+function createToStringArray(innerArray) {
+  return new Proxy([], new ToStringArrayHandler(innerArray));
+}
+```
+
+This solution will work with autotracking in general, since users who access the
+array via `objectAt` will be accessing the tracked property. However, it will
+not integrate with computed property dependencies. If that is needed, then you
+can instead extend Ember's built-in `ArrayProxy` class, which handles forwarding
+events and dependencies itself.
+
+```js
+import ArrayProxy from '@ember/array/proxy';
+import { cached } from 'tracked-toolbox';
+
+class ToStringArray extends ArrayProxy {
+  @cached
+  get _content() {
+    return this.content.map((value) => value.toString());
+  }
+
+  objectAtContent(index) {
+    return this._content.objectAt(index);
+  }
+}
+```
+
+#### Converting code that watches arrays for changes
+
+Array observers and change events can be used to watch arrays and react to
+changes in other ways as well. For instance, you may have a component like
+`ember-collection` which used array observers to trigger a rerender and
+rearrange its own representation of the array. A simplified version of this
+logic looks like the following:
+
+```js
+export default Component.extend({
+  layout: layout,
+
+  init() {
+    this._cells = A();
+  },
+
+  _needsRevalidate(){
+    if (this.isDestroyed || this.isDestroying) {return;}
+    this.rerender();
+  },
+
+  didReceiveAttrs() {
+    this._super();
+
+    this.updateItems();
+  },
+
+  updateItems(){
+    var rawItems = this.get('items');
+
+    if (this._rawItems !== rawItems) {
+      if (this._items && this._items.removeArrayObserver) {
+        this._items.removeArrayObserver(this, {
+          willChange: noop,
+          didChange: '_needsRevalidate'
+        });
+      }
+      this._rawItems = rawItems;
+      var items = A(rawItems);
+      this.set('_items', items);
+
+      if (items && items.addArrayObserver) {
+        items.addArrayObserver(this, {
+          willChange: noop,
+          didChange: '_needsRevalidate'
+        });
+      }
+    }
+  },
+
+  willRender() {
+    this.updateCells();
+  },
+
+  updateCells() {
+    // ...
+  },
+
+  actions: {
+    scrollChange(scrollLeft, scrollTop) {
+      // ...
+      if (scrollLeft !== this._scrollLeft ||
+          scrollTop !== this._scrollTop) {
+        set(this, '_scrollLeft', scrollLeft);
+        set(this, '_scrollTop', scrollTop);
+        this._needsRevalidate();
+      }
+    },
+    clientSizeChange(clientWidth, clientHeight) {
+      if (this._clientWidth !== clientWidth ||
+          this._clientHeight !== clientHeight) {
+        set(this, '_clientWidth', clientWidth);
+        set(this, '_clientHeight', clientHeight);
+        this._needsRevalidate();
+      }
+    }
+  }
+});
+```
+
+We can refactor this to update the cells themselves when they are accessed, by
+accessing them into a computed property that depends on the items array, and
+which updates the cells when it is accessed:
```

Comment:
I found this sentence hard to parse. Can you break the sentence into 2 or 3 instead?